lib/chef/provider/execute.rb

```diff
@@ -80,8 +88,12 @@ def opts
         opts[:umask]       = umask if umask
         opts[:log_level]   = :info
         opts[:log_tag]     = new_resource.to_s
-        if STDOUT.tty? && !Chef::Config[:daemon] && Chef::Log.info? && !sensitive?
-          opts[:live_stream] = STDOUT
+        unless sensitive?
+          if stream_to_formatter?
+            opts[:live_stream] = Chef::EventDispatch::EventsOutputStream.new(run_context.events, :name => :execute)
```

Comment:
Do we want to store a reference to the actual resource object for the future? Seems like the kind of thing that might be useful to formatters.